### ISSUE TITLE
[Bug]: Playwright adds "link://" to anchor tag hrefs, which leads to failing keyboard navigation for WebKit and Mobile Safari

### ISSUE THREAD

**Issue by DrKlonk** [microsoft/playwright#29820]
### Version

1.42.1

### Steps to reproduce

1. Clone my repo at https://gitlab.com/DrKlonk/playwright-adds-link-prefix
2. npm i
3. npm test

### Expected behavior

Test passes for all browsers

### Actual behavior

The test produces different behaviour on different browsers

### Additional context

I want to test keyboard navigation for my app, but it seems WebKit and Mobile Safari skip the links. I presume this is _because_ the link starts with "link://", making it rather useless.

### Environment

```shell
System:
  OS: Windows 11 10.0.22631
  CPU: (8) x64 11th Gen Intel(R) Core(TM) i7-1185G7 @ 3.00GHz
  Memory: 4.21 GB / 15.73 GB
Binaries:
  Node: 18.19.0 - C:\Program Files\nodejs\node.EXE
  npm: 10.2.3 - C:\Program Files\nodejs\npm.CMD
  pnpm: 8.15.2 - C:\Program Files\nodejs\pnpm.CMD 
IDEs:
  VSCode: 1.87.0 - C:\Users\jmi31881\AppData\Local\Programs\Microsoft VS Code\bin\code.CMD
npmPackages:
  @playwright/test: ^1.42.1 => 1.42.1
```

**Comment by dgozman** [comment]
@DrKlonk Thank you for the issue!

Playwright does not actually add `link://` to anchor tags in the page, it only does that in the UI mode snapshot to avoid accidentally navigating to the anchor from the snapshot.

You can try to debug the test to actually see that Firefox and Safari refuse to tab-traverse your example page. Step over a few actions and try to tab around. Let me know whether this works for you.
```ts
npx playwright test --project=firefox --debug
```

I have also manually verified the same by running your http server and opening the page in regular Safari and Firefox. I'd recommend looking into how to make the page tab-traversable, for example adding some `tabindex` attributes.

**Comment by DrKlonk** [comment]
@dgozman Thank you for the reply.

I tried it out and indeed, I cannot navigate to the `<a>` by just using `<kbd>Tab</kbd>`, for both Firefox and Webkit.

However, running the test with or without `--ui` passes the test on Firefox, so I'd like to zoom in on WebKit/Safari.

In "normal" Safari, on any page with anchor tags, pressing `<kbd>Tab</kbd>` works as expected, giving focus and interactivity to the anchor tags.

WebKit in debug mode - or when running the tests - persistently skips over the `<a>`. It seems like the version of WebKit used by Playwright is just not adding `<a>` to the tab-traversal, even though it is an interactive element.

Adding a `tabindex` attribute to `<a>` should not be the solution to this problem, because an anchor is already interactive and should be focusable. Adding the `tabindex="0"` indeed gives the same results.

I really can't imagine skipping `<a>` is default behaviour in WebKit, but I am running out of options on where to look. Do you have any ideas on how to fix this?

I already resigned myself to the option that testing keyboard navigation with Playwright is just not working well enough for my tests.

**Comment by DrKlonk** [comment]
A colleague found the problem. WebKit _does_ skip anchor tags by default: https://github.com/microsoft/playwright/issues/2114. We now use [this fix](https://github.com/microsoft/playwright/issues/2114#issuecomment-1517642401)
